jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactory.java

```diff
@@ -15,29 +15,298 @@
+        }
+    }
+
+    /**
+     * Find a classpath resource.
+     * The {@link Class#getResource(String)} method is used to lookup the resource. If it is not
+     * found, then the {@link Loader#getResource(String)} method is used.
+     * If it is still not found, then {@link ClassLoader#getSystemResource(String)} is used.
+     * Unlike {@link ClassLoader#getSystemResource(String)} this method does not check for normal resources.
+     *
+     * @param resource the relative name of the resource
+     * @return Resource or null
+     */
+    default Resource newClassPathResource(String resource)
+    {
+        URL url = Resource.class.getResource(resource);
+
+        if (url == null)
+            url = Loader.getResource(resource);
+        if (url == null)
+            return null;
+        try
+        {
+            URI uri = url.toURI();
+            return Resource.create(uri);
+        }
+        catch (URISyntaxException e)
+        {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    default Resource newResource(String resource)
+    {
+        return newResource(Resource.toURI(resource));
+    }
+
+    default Resource newResource(Path path)
+    {
+        return newResource(path.toUri());
+    }
+
+    default ResourceCollection newResource(Collection<URI> uris)
+    {
+        return Resource.combine(uris.stream().map(this::newResource).toList());
+    }
+
+    default Resource newResource(URL url)
+    {
+        try
+        {
+            return newResource(url.toURI());
+        }
+        catch (URISyntaxException e)
+        {
+            throw new IllegalArgumentException("Error creating resource from URL: " + url, e);
+        }
+    }
+
+    default Resource newJarFileResource(URI uri)
+    {
+        if (!FileID.isArchive(uri))
+            throw new IllegalArgumentException("Path is not a Java Archive: " + uri);
+        if (!uri.getScheme().equalsIgnoreCase("file"))
+            throw new IllegalArgumentException("Not an allowed path: " + uri);
+        return newResource(URIUtil.toJarFileUri(uri));
+    }
+
+    static ResourceFactory root()
+    {
+        return __ROOT;
+    }
+
+    static ResourceFactory.Closeable closeable()
+    {
+        return new Closeable();
+    }
+
+    static ResourceFactory.LifeCycle lifecycle()
+    {
+        LifeCycle factory = new LifeCycle();
+        org.eclipse.jetty.util.component.LifeCycle.start(factory);
+        return factory;
+    }
+
+    static ResourceFactory of(Resource baseResource)
+    {
+        Objects.requireNonNull(baseResource);
+
+        if (baseResource instanceof ResourceFactory resourceFactory)
+            return resourceFactory;
+
+        return new ResourceFactory()
+        {
+            @Override
+            public Resource newResource(URI resource)
+            {
+                // TODO add an optimized pathway that keeps the URI and doesn't go via String
```

Comment:
mountIfNeeded here? and this resource factory is tracking the mount to close?

Reply:
Done.

Reply:
That's not needed as the returned `ResourceFactory` is working against an already-mounted `Resource` and will only create sub-resources. This cannot end up requiring a new mount.